cinema/tests/test_movie_api.py

```diff
@@ -157,3 +158,179 @@
+
+    def test_filter_movies_by_actor(self):
+        actor_1 = sample_actor()
+        actor_2 = sample_actor()
+
+        movie = sample_movie()
+        movie_actor_1 = sample_movie()
+        movie_actor_2 = sample_movie()
+
+        movie_actor_1.actors.add(actor_1)
+        movie_actor_2.actors.add(actor_2)
+
+        response = self.client.get(
+            MOVIE_URL,
+            {"actors": f"{actor_1.id},{actor_2.id}"}
+        )
+
+        serializer = MovieListSerializer(movie)
+        serializer_actor_1 = MovieListSerializer(movie_actor_1)
+        serializer_actor_2 = MovieListSerializer(movie_actor_2)
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertIn(serializer_actor_1.data, response.data)
+        self.assertIn(serializer_actor_2.data, response.data)
+        self.assertNotIn(serializer.data, response.data)
+
+    def test_retrieve_movie_detail(self):
+        movie = sample_movie()
+        movie.actors.add(sample_actor())
+        movie.genres.add(sample_genre())
+
+        url = detail_url(movie.id)
+
+        response = self.client.get(url)
+
+        serializer = MovieDetailSerializer(movie)
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(response.data, serializer.data)
+
+    def test_create_movie_forbidden(self):
+        payload = {
+            "title": "TestTitle",
+            "description": "TestDescription",
+            "duration": 150,
+        }
+        response = self.client.post(MOVIE_URL, payload)
+
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+
+
+class AdminMovieTests(TestCase):
+    def setUp(self):
+        self.client = APIClient()
+        self.user = get_user_model().objects.create_user(
+            email="admin@admin.admin",
+            password="admin_password",
+            is_staff=True
+        )
+        self.client.force_authenticate(self.user)
+
+    def test_create_movie(self):
+        actor_1 = sample_actor()
+
+        genre_1 = sample_genre(name="Action")
+
+        payload = {
+            "title": "TestTitle",
+            "description": "TestDescription",
+            "duration": 150,
+            "actors": [actor_1.id],
+            "genres": [genre_1.id],
+        }
+        response = self.client.post(MOVIE_URL, payload)
+
+        movie = Movie.objects.get(id=response.data["id"])
+        actors = movie.actors.all()
+        genres = movie.genres.all()
+
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+
+        self.assertIn(actor_1, actors)
+        self.assertEqual(actors.count(), 1)
+        self.assertIn(genre_1, genres)
+        self.assertEqual(genres.count(), 1)
+
+
+    def test_delete_movie_not_allowed(self):
+        movie = sample_movie()
+        response = self.client.delete(detail_url(movie.id))
+
+        self.assertEqual(
+            response.status_code,
+            status.HTTP_405_METHOD_NOT_ALLOWED
```

Comment:
This test case is not clear. It's better to give a meaningful name to the test case that describes what it is testing. For example, it could be `test_delete_movie_is_not_allowed` or something similar.